extensions/sql-database-projects/src/models/tree/fileFolderTreeItem.ts

```diff
@@ -87,15 +88,18 @@ export function sortFileFolderNodes(a: (FolderNode | FileNode), b: (FolderNode |
 /**
  * Converts a full filesystem URI to a project-relative URI that's compatible with the project tree
  */
-function fsPathToProjectUri(fileSystemUri: vscode.Uri, projectNode: ProjectRootTreeItem): vscode.Uri {
+function fsPathToProjectUri(fileSystemUri: vscode.Uri, projectNode: ProjectRootTreeItem, isFile?: boolean): vscode.Uri {
 	const projBaseDir = projectNode.project.projectFolderPath;
 	let localUri = '';
 
 	if (fileSystemUri.fsPath.startsWith(projBaseDir)) {
 		localUri = fileSystemUri.fsPath.substring(projBaseDir.length);
 	}
-	else {
-		throw new Error(`Project (${projBaseDir}) pointing to file outside of directory (${fileSystemUri.fsPath})`);
+	else if (isFile) {
+		// if file is outside the folder add add at top level in tree
+		// this is not true for folders otherwise the outside files will not be directly inside the top level
+		let parts = utils.getPlatformSafeFileEntryPath(fileSystemUri.fsPath).split('/');
+		localUri = parts[parts.length - 1];
```

Comment:
What's the intended behavior if I have a folder outside the project directory that contains files of its own?  Will the files be inside the folder that's located at the tree's top level, or will there be an empty folder alongside files all at the top level?

```
MyProj
    /OutsideFolder
        /OutsideFileInOutsideFolder.sql
```

or

```
MyProj
    /OutsideFolder
    /OutsideFileInOutsideFolder.sql
```

Reply:
MyProj
    /OutsideFileInOutsideFile.sql
    /OutsideFileInOutsideFile2.sql

(no outside folders in tree - as per SSDT)